src/compiler/chainrules.jl

```diff
@@ -10,13 +13,17 @@ If it does not, then the second argument is a list of edges to attach to the Cod
 such that if a suitable rule is defined later, the generated function will recompile.
 """
 function has_chain_rrule(T)
-  m = meta(Tuple{typeof(rrule),T.parameters...})
-  if m.method !== chainrules_fallback
+  return_type = Core.Compiler.return_type(rrule, T)
```

Comment:
Looks like you are trying to use `Core.Compiler.return_type` *inside* the generated function. That is not what you want: call this outside the the generated function, and possibly pass the answer into the generated function if necessary.

Reply:
I am not sure that is possible.
There is no outside the generated functions.
Within the 
`_pullback` generated function is the first place we have the information that is needed to decide if we should use a ChainRule or not.

Reply:
Is the problem that the content of the generated fuynction *always* runs in the world-age where it was declared.
Not in the world-age where it was run to generate the function?
So backedges can't fix it, as they don't change when it was declared they just cause it to recompile, but it still acts in that world age where it was declared regardless?

Reply:
Calling `Core.Compiler.return_type` inside `@generated` should be entirely prohibited, as it also risks crashing the compiler, but I am not yet brave enough to address that problem

Reply:
ok, I will find another solution then.